src/main.rs

```diff
@@ -18,10 +19,13 @@ async fn main() -> eyre::Result<()> {
     tracing::info!("REST API on http://{}", listener.local_addr()?);
     tracing::info!("API docs on http://{}/api-docs", listener.local_addr()?);
 
-    let app_state = app_state::AppState::init().await?;
+    let secret = auth::init_secret()?;
+    let app_state = state::AppState::init().await?;
+
+    let _ = auth::encode_token(&secret)?;
```

Comment:
```suggestion
    let app_state = state::AppState::init().await?;
    
    let secret = auth::init_secret()?;
    let _ = auth::encode_token(&secret)?;
```

Nit: secret is not needed to init app_state

Reply:
updated but i'm leaving the initialization in the order the variables appear in the signature of the `rest::app` call